### PR TITLE
Build native extensions on M1

### PR DESCRIPTION
**Pull request by bbuchalter**
We already support disabling this compilation flag anyway, so might as well just do it automatically for architectures that don't support it.

### CODE REVIEW

**Comment by bbuchalter** [comment]
Hello, just a friendly poke.

Based on https://github.com/sass/sassc-ruby/graphs/contributors perhaps @bolandrm or @eregon could help?

**Comment by bbuchalter** [comment]
Friendly ping here. 👋

**Comment by bbuchalter** [comment]
Maybe @glebm or @Edward could help?

File: ext/extconf.rb [hunk 1]
```diff
@@ -19,7 +19,9 @@
   $LDFLAGS << ' -static-libgcc -static-libstdc++'
 end
 
-if enable_config('march-tune-native', false)
+# darwin arm doesn't support native
+# see: https://github.com/sass/sassc-ruby/issues/222
+if enable_config('march-tune-native', false) && Gem::Platform.local.cpu != 'arm64'
```
Comment (eregon) [inline ext/extconf.rb:24]: The default is already `false`, why is this change needed?

**Comment by bbuchalter** [comment]
Upon further investigation, this issue only exists in versions prior to v2.3.0. It's not clear which commit fixes it, but this does address this error:
```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /Users/brian.buchalter/.gem/ruby/2.7.4/gems/sassc-2.2.1/ext
/Users/brian.buchalter/.rubies/ruby-2.7.4/bin/ruby -I /Users/brian.buchalter/.rubies/ruby-2.7.4/lib/ruby/2.7.0 -r ./siteconf20210720-81997-w05u9w.rb extconf.rb
creating Makefile

current directory: /Users/brian.buchalter/.gem/ruby/2.7.4/gems/sassc-2.2.1/ext
make "DESTDIR=" clean

current directory: /Users/brian.buchalter/.gem/ruby/2.7.4/gems/sassc-2.2.1/ext
make "DESTDIR="
compiling ./libsass/src/units.cpp
clang: error: the clang compiler does not support '-march=native'
make: *** [units.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/brian.buchalter/.gem/ruby/2.7.4/gems/sassc-2.2.1 for inspection.
Results logged to /Users/brian.buchalter/.gem/ruby/2.7.4/extensions/arm64-darwin-20/2.7.0-static/sassc-2.2.1/gem_make.out

An error occurred while installing sassc (2.2.1), and Bundler cannot continue.
Make sure that `gem install sassc -v '2.2.1'` succeeds before bundling.
```

**Comment by bbuchalter** [comment]
Per your comment, it was probably https://github.com/sass/sassc-ruby/pull/158
